### ISSUE TITLE
@onDismissed event of <BsAlert> is called twice

### ISSUE THREAD

**Issue by jelhan** [kaliber5/ember-bootstrap#1264]
The action bound to `@onDismissed` event of `<BsAlert>` is called twice. Both if dismissed using the rendered button and if the property bound to `@visible` argument is set to `false`.

## Reproduction

1. Create a new ember application and install latest version of Ember Bootstrap (v4.1.2).
2. Create this application template:
    ```hbs
    <button {{on "click" this.hideAlert}}>
      hide
    </button>

    <BsAlert
      @onDismissed={{this.hideAlert}}
      @visible={{this.showAlert}}
    >
      an alert
    </BsAlert>
    ```
3. Create this application controller:
    ```js
    import Controller from '@ember/controller';
    import { action } from '@ember/object';
    import { tracked } from '@glimmer/tracking';

    export default class ApplicationController extends Controller {
      @tracked showAlert = true;

      @action
      hideAlert() {
        console.log('hideAlert');
        this.showAlert = false;
      }
    }
    ```

I expected to see `hideAlert` console message only once if alert is dismissed. But it's shown twice in console.

**Comment by ijlee2** [comment]
Hi, @jelhan. I'm not familiar with the addon yet, but would like to see if I can help with this issue for Hacktoberfest. Can you assign the issue to me?

**Comment by jelhan** [comment]
Sure. Thanks a lot for supporting us. Feel free to reach out to me if you have any question.

**Comment by ijlee2** [comment]
Thanks!

**Comment by jelhan** [comment]
The reproduction in the initial post is wrong if using the external button to hide the alert and not the dismiss button rendered by the component itself.

The external button is using the same method as click event handler on the button (`{{on "click" this.hideAlert}}`) as well as on `@onHidden={{this.hideAlert}}` argument. In that case I would actually expect that `hideAlert` is called twice if the button clicked.

A simplified reproduction is this one:

```hbs
<BsAlert
  @onDismissed={{this.onHidden}}
  @visible={{this.showAlert}}
>
  an alert
</BsAlert>
```

```js
import Controller from '@ember/controller';
import { action } from '@ember/object';
import { tracked } from '@glimmer/tracking';

export default class ApplicationController extends Controller {
  @tracked showAlert = true;

  @action
  hideAlert() {
    console.log('hideAlert');
    this.showAlert = false;
  }
}
```

A click on the dismiss button causing the alert to be hidden, which executes the `hideAlert` method passed as `@onDismissed` argument. `hideAlert` method sets `showAlert` to `false`, which is passed to the alert as `@visible` argument. This causes the alert to be hidden again (even though it's already hidden), which executes the `hideAlert` method again. We have an endless loop.

The issue can be reproduced on `master` but is fixed by #1281.
